app/scaffold-controllers/navigationController.js

```diff
@@ -11,17 +13,26 @@ function(
     Application,
     AnchoredLayoutPlugin,
     NavigationPlugin,
-    NavigationHeaderController
+    NavigationHeaderController,
+    SearchBarController,
+    SearchConfig
 ) {
 /* eslint-enable */
-    var NavigationController = function(tab, layout, navigationView, navigationHeaderController, includeDrawerIcon) {
+    var NavigationController = function(tab, layout, navigationView, navigationHeaderController, searchBarController, includeDrawerIcon) {
         this.id = tab.id;
         this.isActive = false;
         this.viewPlugin = layout;
         this.navigationView = navigationView;
         this.navigationHeaderController = navigationHeaderController;
+        this.searchBarController = searchBarController;
 
         this.navigate(tab.url, includeDrawerIcon);
+
+        var self = this;
+
+        this.searchBarController.registerSearchSubmittedEvents(function(params) {
+            self.navigate(self.searchBarController.generateSearchUrl(params.searchTerms));
```

Comment:
Personal preference, but maybe capture `self.searchBarController.generateSearchUrl(params.searchTerms)` into a variable for readability?
I think it helps with self documentation.